source/particle/world.cc

```diff
@@ -54,10 +54,9 @@ namespace aspect
     {
       CitationInfo::add("particles");
       if (particle_load_balancing & ParticleLoadBalancing::repartition)
-#if DEAL_II_VERSION_GTE(9,4,0)
+
         this->get_triangulation().signals.weight.connect(
-#else
-        this->get_triangulation().signals.cell_weight.connect(
+
 #endif
```

Comment:
If you remove the `#if` and the `#else`, then you also need to remove the `#endif`.

Same below in all of the other parts of your patch.